index.js

```diff
@@ -87,6 +87,8 @@ is.generator = x => is.iterable(x) && is.function(x.next) && is.function(x.throw
 // TODO: Change to use `isObjectOfType` once Node.js 6 or higher is targeted
 is.generatorFunction = x => is.function(x) && is.function(x.constructor) && x.constructor.name === 'GeneratorFunction';
 
```

Comment:
you see that `x => is.function(x) && is.function(x.constructor) && x.constructor.name === 'PASS PARAM HERE';`

happens twice. Now, I don't precisely know how to name that method, but this should definitely get extracted to a method.

Reply:
Added `isFunctionOfType` method.